### ISSUE TITLE
PLIP: New feature hide and show Volto blocks

### ISSUE THREAD

**Issue by SaraBianchi** [plone/volto#6559]
## Abstract

New feature to hide or show blocks in view from the editor editing mode.

## Motivation

Some customers have expressed the need to set up temporary blocks that will be activated or deactivated as needed.
A real use case could be to set up a block with a link to a YouTube live video that will be activated only when there is a session/conference.

## Proposal & Implementation

In the Volto EditBlockWrapper, a button with an icon (eye/crossed-out eye) has been added to the center and top of the wrapper to hide or show a certain block. In addition to this, a button has also been added to the left side toolbar (above the Undo/Redo buttons) to switch to "Show hidden blocks too" mode or to collapse them to make the page preview functional.
More wrap layout details on the pull request.

## Deliverables

The feature was requested in Volto version 17, but if it works we may open a pr for Volto main as well. Other packages were not affected.
This feature is quite urgent.

## Risks

As explained in the pr, there is a risk of getting stuck inside collapsed blocks if you press the DEL button while moving from the keyboard, possible improvements can be managed in Slate type blocks, where it will be necessary to specify that if there are previously hidden blocks they must be bypassed.

## Participants

Piero Nicolli [@pnicolli](https://github.com/pnicolli) participated in the review and exposition of the functionality already proposed in the past by me (@SaraBianchi).

## Pull Request

Old closed pr: #5157
Open pr: #6556

**Comment by tisto** [comment]
The feature is valuable, and something clients asked us to do. So far, we refused to implement this feature because we see too many UX issues that have no obvious solution. The main problem, IMHO, is that the edit and the view view start to differ, which can confuse editors. To gain more experience with this, I'd recommend adding this feature with a feature flag first and gaining some experience and feedback before we enable it. This is usually our default approach when we change the Volto user interface.

**Comment by sneridagh** [comment]
I agree with Timo's point of view, let's discuss it more in depth in the Volto Team meeting and let the PLIP goes through the process.

**Comment by stevepiercy** [comment]
@djay did you see [Plone Improvement Proposals (PLIPs)](https://6.docs.plone.org/contributing/core/plips.html)?

**Comment by djay** [comment]
Yes.

**Comment by stevepiercy** [comment]
In that case, can you suggest how to improve this documentation so that addresses your concern?

https://6.docs.plone.org/contributing/core/plips.html#designated-teams

> ## Designated teams
> The following packages and repositories have designated teams that you can contact by `@`-ing them in its GitHub issue tracker. If the repository is not listed, use the default repository for `Products.CMFPlone`.
> 
> Volto [plone/volto](https://github.com/plone/volto) `@plone/volto-team`

**Comment by djay** [comment]
There is stuff about soliciting feedback which is very vague. In mentions both the community adn the team but it's not clear which and when and what you are asking and how long to ask on what kind of approval you are seeking.
IMO there should always feedback from UX experts on anything that toches the UI but how is anyone supposed to know who is a UX expert?

Then there is this concept of submitting for approval. I know through experience what this means is the author needs to put the PLIP in the agenda of the team meeting and go there and defend their PLIP. Which any team member knows. But is not in the documentation and is now not possible because agendas are no longer public.
 It talks about "resubmits the PLIP" but I'm not sure what that means. delete the ticket and recreate it?

Then it talks about final approval. "you can request a final review and approval". How? you can request a review of a PR but not a ticket. is this meant to be done by @ the team? if so say so.

Take one step back and really think about this process from the point of view of someone who has not been to a volto roadmap discussion and had their PLIP idea preblessed and understand exactly how vague and non-transparent this process is.

**Comment by stevepiercy** [comment]
@djay to avoid hijacking this issue, please submit an issue for or PR with any specific change that would improve the documentation of the PLIP process in `plone/documentation`. PRs are better, IMO. Thank you!
